internal/profile/install_profile.go

```diff
@@ -0,0 +1,202 @@
+		}
+		profiles = append(profiles, metadata)
+	}
+	return profiles, nil
+}
+
+// createProfile installs a new profile at the given package path location.
+// overwriteExisting determines the behavior if a profile with the given name already exists.
+// On true, it'll overwrite the profile, on false, it'll copy the existing profile over to profileName_old
+func createProfile(options Options) error {
+	profile, err := NewConfigProfile(options.PackagePath, options.Name)
+	if err != nil {
+		return errors.Wrap(err, "error creating profile")
+	}
+
+	// check to see if we have an existing profile at that location.
+	exists, err := profile.alreadyExists()
+	if err != nil {
+		return errors.Wrap(err, "error checking for existing profile")
+	}
+	if exists {
+		localChanges, err := profile.localFilesChanged()
+		if err != nil {
+			return errors.Wrapf(err, "error checking for changes in %s", profile.ProfilePath)
+		}
+
+		// If there are changes and we've selected CreateNew, move the old path
+		if localChanges && !options.OverwriteExisting {
+			if localChanges && options.Name == DefaultProfile {
+				logger.Warn("default profile has been changed by user or updated by elastic-package. The current profile will be moved.")
+			}
+			err = updateExistingDefaultProfile(options.PackagePath)
+			if err != nil {
+				return errors.Wrap(err, "error moving old profile")
+			}
+			err = os.Mkdir(profile.ProfilePath, 0755)
+			if err != nil {
+				return errors.Wrapf(err, "error crating profile directory %s", profile.ProfilePath)
+			}
+		}
+	} else {
+		err = os.Mkdir(profile.ProfilePath, 0755)
+		if err != nil {
+			return errors.Wrapf(err, "error crating profile directory %s", profile.ProfilePath)
+		}
+	}
+	if err != nil {
+		return errors.Wrapf(err, "stat file failed (path: %s)", profile.ProfilePath)
+	}
+
+	// write the resources
+	err = profile.writeProfileResources()
+	if err != nil {
+		return errors.Wrap(err, "error writing profile file")
+	}
+	return nil
+}
+
+// updateExistingDefaultProfile migrates the old default profile to profile_old
+func updateExistingDefaultProfile(path string) error {
+	profile, err := NewConfigProfile(path, DefaultProfile)
+	if err != nil {
+		return errors.Wrap(err, "error creating profile")
+	}
+	meta, err := profile.metadata()
+	if err != nil {
+		return errors.Wrap(err, "error updating metadata")
+	}
+	newName := fmt.Sprintf("default_%s_%d", meta.Version, meta.DateCreated.Unix())
+	newFilePath := filepath.Join(filepath.Dir(profile.ProfilePath), newName)
+	meta.Name = newName
+	meta.From = newFilePath
+
+	err = profile.updateMetadata(meta)
+	if err != nil {
+		return errors.Wrap(err, "error updating metadata")
+	}
+
+	err = os.Rename(profile.ProfilePath, newFilePath)
+	if err != nil {
+		return errors.Wrap(err, "error moving default profile")
+	}
+
+	return nil
+}
+
+// createProfileFrom creates a new profile by copying over an existing profile
+func createProfileFrom(option Options) error {
```

Comment:
I have feeling that this function is similar to `createProfile`. Is it possible to unify it somehow?

Reply:
Yah, good idea.